src/components/layouts/app-layout.tsx

```diff
@@ -1,29 +1,50 @@
 import React from "react";
 import { useAppDispatch, useAppSelector } from "../../app/hooks";
-import { AppBar, Box, Drawer, Toolbar, Typography, Select, MenuItem } from "@material-ui/core";
+import { AppBar, Box, Drawer, Toolbar, Typography, Select, MenuItem, Button } from "@material-ui/core";
 import useAppLayoutStyles from "styles/layouts/app-layout";
 import siteLogo from "../../gfx/Metatavu-icon.svg";
 import strings from "localization/strings";
+import { Link } from "react-router-dom";
 import { selectLocale, setLocale } from "features/locale/locale-slice";
+import classNames from "classnames";
+import jwt_decode from "jwt-decode";
+import { logout, selectAuth } from "features/auth/auth-slice";
+import theme from "theme/theme";
 
 /**
  * Component properties
  */
 interface Props {
-  drawerContent: React.ReactNode;
-  editorContent: React.ReactNode;
+  drawerContent?: React.ReactNode;
+  children: React.ReactNode;
+  managementScreen?: boolean
 }
 
 /**
  * Application layout component
  *
  * @param props component properties
  */
-const AppLayout: React.VoidFunctionComponent<Props> = ({ drawerContent, editorContent }) => {
+const AppLayout: React.VoidFunctionComponent<Props> = ({ drawerContent, children, managementScreen }) => {
   const classes = useAppLayoutStyles();
   const dispatch = useAppDispatch();
+  const { accessToken } = useAppSelector(selectAuth);
+  const [ isAdmin, setIsAdmin ] = React.useState(false);
   const { locale } = useAppSelector(selectLocale);
 
+  React.useEffect(() => {
+    if (!accessToken) {
+      return;
+    }
+
+    const decodedToken: any = jwt_decode(accessToken.access_token);
+    const roles = decodedToken.realm_access.roles as string[];
+
+    if (roles.includes("admin")) {
+      setIsAdmin(true);
+    }
```

Comment:
No need to use jwt_decode since AccessToken type already has the needed information (roles + other stuff).